### DISCUSSION TITLE
Can I prevent Vitest creating JSON files in a `coverage` directory whenever tests run?

### DISCUSSION THREAD

**Discussion by jameshibbard** [vitest-dev/vitest#1897] [Q&A] [answered]
Hi,

I'm using Vitest in conjunction with Vue Test Utils to test Vue components in a Rails app. 
Whenever I run Vitest, it outputs a bunch of `.json` files to a `coverage` folder in my `test` directory.
These JSON files all have the format `coverage-3082544-1661159062465-7.json` etc.

Is there any way to stop Vitest creating these files?

I'm configuring Vitest from within a `vite.config.ts` file at the root of my app.

```js
export default defineConfig({
  plugins: [ ... ],
  test: {
    environment: 'jsdom',
    include: ['../../test/javascript/components/*.test.js'],
    coverage: {
      reportsDirectory: '../../test/javascript/coverage',
    },
  },
});
```

I know I've defined a `reportsDirectory`. If I don't do that, Vitest creates it anyway in an adjacent folder to my entry point.

Grateful for any help.

**Comment by jameshibbard** [comment, marked as answer]
I just upgraded to v 0.23.1 and Vitest no longer creates a `coverage` directory.

**Reply by daribock** [reply]
I am on v0.23.2 and it still creates the folder. Am I doing something wrong? This is my `vite.config.ts`:
```Code
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import eslintPlugin from 'vite-plugin-eslint';
import i18nHotReload from './plugins/i18n-reload';

export default defineConfig((configEnv) => ({
    plugins: [react(), eslintPlugin(), i18nHotReload()],
    server: {
        port: 3000,
    },
    test: {
        globals: true,
        environment: 'jsdom',
        include: ['src/**/*.test.{ts,tsx}'],
        coverage: {
            reporter: ['text'],
        },
    },
}));
```
